### DISCUSSION TITLE
Substring a String value

### DISCUSSION THREAD

**Discussion by abhishekrongali** [schibsted/jslt#316] [General]
Hi,

Can we substring a string based on size? 

Example:

1. let a = "123456678" ->   I want to split this value based on size and needed first 4 characters only
 Expected value = "1234"

**Comment by catull** [comment]
@abhishekrongali Have you already consulted this [page](https://github.com/schibsted/jslt/blob/master/functions.md) ?

@larsga 

I have tried to come up with some approaches.

I have this input

```
{
  "s": "1234567890"
}
```

This is the transformation.

```
{
  "s": .s,
  "capture": capture (.s, "^(?<first>.{4})(?<rest>.*)$"),
  "first": .capture.first,
  "replace": replace (.s, "^(.{4}).*$", "$1")
}
```

And this is the result:

```
{
  "s" : "1234567890",
  "capture" : {
    "first" : "1234",
    "rest" : "567890"
  },
  "first" : "1234",
  "replace" : "$1"
}
```

Now, unless I am doing something wrong, the `replace` method should allow for regexp references, similar to the capture case.
I have tried with "%0", "_0", "$0", as well as "%1", "_1", "$1", and a few other; none of them work.

It should work, see this online [example](https://regex101.com/r/pVeWam/1) I created quickly.

**Reply by abhishekrongali** [reply]
Thank you, works perfectly! :)
